Processing Mario's feedback on Infrastructure & Software Requirements (#124)
CLARIAH/clariah-plus: requirements/infrastructure-requirements.md

so the infrastructure must check expirations and renew certificates automatically,
for instance using [Let's Encrypt](https://letsencrypt.org).

### 13. The infrastructure *MUST* back up all application data from deployed services at an agreed upon interval. ([REL](#rel))

Data loss is unacceptable, especially when that data has been provided by users in
[stateful applications](#3-the-infrastructure-must-be-able-to-run-_stateful-applications_).
Both backup and restore functionality *MUST* be tested.

### 14. The infrastructure *MUST* be _GDPR-compliant_, for instance in the way it stores data.

If the infrastructure stores personal data, for instance logs, this must be done in a GDPR-compliant manner.

> This entails, amongst others, that the infrastructure *SHOULD* give users the
> option to download all their user data and *MUST* give users the option to
> remove their data.

Reviewer: While this point was already in the Infrastructure Requirements, I’m no longer sure if this is an infrastructural concern. Isn’t it up to the software applications to offer inspection/removal functionality? Of course, we still want to prevent that the infrastructure needlessly logs any PII and, if it does need to capture it, offers removal functionality to the applications that run on the infrastructure.

```suggestion
The infrastructure *MUST* store personally identifiable information (PII) only if necessary. 
If the infrastructure stores PII, it *MUST* give users (or software applications on behalf of users) the option to remove their data
and it *SHOULD* give users the option to download their data.
```

Author: I agree, that sounds clearer indeed!

All software components, including the infrastructure’s OS and other packages, *MUST* be continuously updated to incorporate security patches.

### 16. The infrastructure *MUST* store, manage and deliver secrets and other sensitive data to application ([REL](#rel))

The infrastructure authenticates, validates, authorizes and grants access to
clients for accessing secrets (user names, passwords, API tokens) that are
needed to use or run services within the Infrastructure, these *MUST NOT* be
passed as unencrypted environment variables (see also point 4) as those are by nature
prone to leak and become compromised.

> Solutions *MAY* include [kubernetes Vault](https://github.com/hashicorp/vault-k8s) or similar software.

Reviewer: Just to make clear what _category_ of solutions we’re talking about here:

```suggestion
Solutions *MAY* include a secrets store such as [kubernetes Vault](https://github.com/hashicorp/vault-k8s).
```
